runtime/queries/metricsview.go

```diff
@@ -292,6 +292,327 @@ func buildFilterClauseForCondition(mv *runtimev1.MetricsViewSpec, cond *runtimev
+	return identifier{safeName(name), false}
+}
+
+func dimensionAliases(mv *runtimev1.MetricsViewSpec) map[string]identifier {
+	aliases := map[string]identifier{}
+	for _, dim := range mv.Dimensions {
+		aliases[dim.Name] = identifier{safeName(metricsViewDimensionColumn(dim)), dim.Unnest}
+	}
+	return aliases
+}
+
+func buildFromExpression(expr *runtimev1.Expression, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
+	var emptyArg []any
+	switch e := expr.Expression.(type) {
+	case *runtimev1.Expression_Val:
+		arg, err := pbutil.FromValue(e.Val)
+		if err != nil {
+			return "", emptyArg, err
+		}
+		return "?", []any{arg}, nil
+
+	case *runtimev1.Expression_Ident:
+		col, ok := allowedIdentifiers[e.Ident]
+		if !ok {
+			return "", emptyArg, fmt.Errorf("unknown column filter. %s", e.Ident)
+		}
+		return col.mapped, emptyArg, nil
+
+	case *runtimev1.Expression_Cond:
+		return buildFromConditionExpression(e.Cond, allowedIdentifiers, dialect)
+	}
+
+	return "", emptyArg, nil
+}
+
+func buildFromConditionExpression(cond *runtimev1.Condition, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
+	switch cond.Op {
+	case runtimev1.Operation_OPERATION_LIKE, runtimev1.Operation_OPERATION_NLIKE:
+		return buildFromLikeExpression(cond, allowedIdentifiers, dialect)
+
+	case runtimev1.Operation_OPERATION_IN, runtimev1.Operation_OPERATION_NIN:
+		return buildFromInExpression(cond, allowedIdentifiers, dialect)
+
+	case runtimev1.Operation_OPERATION_AND:
+		return buildFromAndOrExpressions(cond, allowedIdentifiers, dialect, " AND ")
+
+	case runtimev1.Operation_OPERATION_OR:
+		return buildFromAndOrExpressions(cond, allowedIdentifiers, dialect, " OR ")
+
+	default:
+		leftExpr, args, err := buildFromExpression(cond.Exprs[0], allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+
+		rightExpr, subArgs, err := buildFromExpression(cond.Exprs[1], allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+		args = append(args, subArgs...)
+
+		return fmt.Sprintf("(%s) %s (%s)", leftExpr, conditionExpressionOperation(cond.Op), rightExpr), args, nil
+	}
+}
+
+func buildFromLikeExpression(cond *runtimev1.Condition, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
+	if len(cond.Exprs) != 2 {
+		return "", nil, fmt.Errorf("like/not like expression should have exactly 2 sub expressions")
+	}
+
+	leftExpr, args, err := buildFromExpression(cond.Exprs[0], allowedIdentifiers, dialect)
+	if err != nil {
+		return "", nil, err
+	}
+
+	rightExpr, subArgs, err := buildFromExpression(cond.Exprs[1], allowedIdentifiers, dialect)
+	if err != nil {
+		return "", nil, err
+	}
+	args = append(args, subArgs...)
+
+	notKeyword := ""
+	if cond.Op == runtimev1.Operation_OPERATION_NLIKE {
+		notKeyword = "NOT"
+	}
+
+	// identify if immediate identifier has unnest
+	// TODO: do we need to do a deeper check?
```

Comment:
Not sure, but I don't think so? If we wanted to stay simple, might even be okay to make the assumption that the first arg to an operation is an ident